src/plccng/scanner/scanner_test.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+
+from ..lines import parseLines
+from .scanner import Scanner
+from .structs import LexError, Line, Skip, Token
+
+
+@pytest.fixture
+def errorRaisingMatcher():
+    class ErrorRaisingMatcher:
+        def match(self, line, index):
+            return LexError(line=line, column=index)
+
+    return ErrorRaisingMatcher()
+
+@pytest.fixture
+def whiteSpaceSkipMatcher():
+    class WhiteSpaceSkipMatcher():
+        def match(self, line, index):
+            if line.string[index] == " ":
+                return Skip(name="whitespace", lexeme=" ", line=line, column=index)
+            else:
+                return LexError(line=line, column=index)
+    
+    return WhiteSpaceSkipMatcher()
+
+#Matches up to 5 of any characters (whitespace included) in a string
+@pytest.fixture
+def fiveCharacterMatcher():
+    class FiveCharacterMatcher():
+        def match(self, line, index):
+            final_index = len(line.string) - 1
+            if index+5 > final_index:
+                return Token(name="Token", lexeme=line.string[index:final_index], line=line, column=final_index)
+            return Token(name="Token", lexeme=line.string[index:index+5], line=line, column=index+5)
+    
+    return FiveCharacterMatcher()
+
+    
+
+
+def test_no_lines_given_returns_nothing():
+    scanner = Scanner(matcher=None)
+    results = scanner.scan(None)
+    assert list(results) == []
+
+
+def test_empty_lines_given_returns_nothing():
+    scanner = Scanner(matcher=None)
+    results = scanner.scan([])
+    assert list(results) == []
+
+
+def test_LexError(errorRaisingMatcher):
+    scanner = Scanner(errorRaisingMatcher)
+    oneLineWithError = parseLines('''\
+blah blah
+''')
+    result = list(scanner.scan(oneLineWithError))
+    assert isinstance(result[0], LexError)
+
+
+def test_LexError_at_the_start_of_the_line_moves_on_to_next_line(errorRaisingMatcher):
+    scanner = Scanner(errorRaisingMatcher)
+    twoLinesWithErrors = parseLines('''\
+blah blah
+next line
+''')
+    scanner = Scanner(errorRaisingMatcher)
+    results = list(scanner.scan(twoLinesWithErrors))
+    assert all(isinstance(item, LexError) for item in results)
```

Comment:
Prove that two were scanned. This assert would pass even if there were only one.